### ISSUE TITLE
'ignorePattern' issue with modifier

### ISSUE THREAD

**Issue by ulrichb** [jonaskello/eslint-plugin-functional#45]
Using the following config:

`"functional/prefer-readonly-types": ["error", { ignorePattern: ["^mutable"] }]`

... the following code reports an error.

```TS
export class SomeClass {
    mutableSomethingA?: string;
    private mutableSomethingB?: string; // False positive "A readonly modifier is required"
}
```

Workaround:
`"functional/prefer-readonly-types": ["error", { ignorePattern: ["^(private )?mutable"] }]`

**Comment by ulrichb** [comment]
Similar issue with locals:

```ts
const mutableResult: string[] = [];
```

**Comment by RebeccaStevens** [comment]
As of right now, we haven't sorted out what text in each node type for each rule "ignorePattern" should check against. We'll make sure this is sorted before the 1.0 release.

**Comment by ulrichb** [comment]
Okay. Thanks for the update!

**Comment by ulrichb** [comment]
Similar issue:

```ts
delete mutableLocal[mySymbol];
```

...cannot be 

ignored with `ignorePattern: ["^mutable"]`

Workaround: `ignorePattern: ["^(delete )?mutable"]`

**Comment by ulrichb** [comment]
_Maybe_ similar issue:
```ts
return Object.assign(mutableLocal, { prop: 1 });
```

Here I found no workaround.

**Comment by ulrichb** [comment]
_Maybe_ related issue:

```ts
class SomeClass {
    mutableProp1: readonly string[] = []; // => no error

    readonly mutableProp2: string[] = []; // => functional/prefer-readonly-type
}
```

So the suppression works for the `readonly` of the property itself, but does not get applied for the property's _type_ (non readonly `string[]`).

**Comment by ulrichb** [comment]
So these are all my scenarios which (I think) worked in the TSLint version and not in `eslint-plugin-functional`.

**Comment by RebeccaStevens** [comment]
@ulrichb Thanks for finding all these :)

**Comment by ulrichb** [comment]
Many thanks @RebeccaStevens!

**Comment by RebeccaStevens** [comment]
I'm about to make a new release. You can test that it's all working if you want :)

**Comment by ulrichb** [comment]
Yeah! Will indeed test it :)

**Comment by ulrichb** [comment]
Hi. Just tested it.

`Object.assign`, `delete` and `Array` type works.

... but I have now a new error for this case:

```ts
const mutableArr: string[] = [];
mutableArr.push(""); // => functional/immutable-data
```
(worked in 0.5.1)

**Comment by ulrichb** [comment]
... same issue with `mutableArr.splice(...)`

**Comment by RebeccaStevens** [comment]
Good spot. I know exactly what's going on with this, I'll get it fixed up soon.
Do you want to open a new issue for this?

Looks like the tests are a little incomplete 😥
